x/incentives/keeper/distribute_test.go

```diff
@@ -545,62 +543,102 @@ func (suite *KeeperTestSuite) TestNoLockNonPerpetualGaugeDistribution() {
 	suite.Require().Equal(gauges[0].String(), expectedGauge.String())
 }
 
-func (suite *KeeperTestSuite) TestGetValidConcentratedLiquidityGauge() {
+func (suite *KeeperTestSuite) TestGetPoolFromGaugeId() {
+	const (
+		validGaugeId = uint64(1)
+		validPoolId  = validGaugeId
+	)
+
 	tests := []struct {
-		name              string
-		gaugeId           uint64
-		expectedGaugeType poolmanagertypes.PoolType
-		expectErr         bool
+		name    string
+		gaugeId uint64
+		// For balancer pools, we do not create this link after pool
+		// creation. As a result, for edge case testing we want
+		// to manually set the link.
+		shouldSetPoolGaugeId bool
+		// this flag is necessary for edge case test where
+		// there is a link between gauge and pool but the pool
+		// does not exist.
```

Comment:
It is not. Expanded comment to also say:

```
This case should not happen
in practice because the gauges must be created
after pool creation, via hook. However, we test
it for coverage.
```